### ISSUE TITLE
Text to Speech Screen Reader on MacOS doesn't read selected text.

### ISSUE THREAD

**Issue by x0054** [warpdotdev/Warp#6650]
### Pre-submit Checks

- [x] I have [searched Warp bugs](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3ABugs) and there are no duplicates
- [x] I have [searched Warp known issues page](https://docs.warp.dev/help/known-issues) and my issue is not there
- [ ] I have included the logs (optional, but helps expedite the bug fix). [Log gathering intructions](https://docs.warp.dev/support-and-billing/sending-us-feedback#gathering-warp-logs)

### Describe the bug

On MacOS 15.5 when I select response from the LLM or the terminal and hit a keyboard shortcut to read the text, it reads the entire screen rather than actually selected text.

### To reproduce

1. First, turn on the feature on your computer. To do so, go to MacOS Settings, in Settings, search for Speak Selection. Click the little toggle, and next to the toggle, hit on the round eye icon. That should open an additional dialog box. In the dialog box, check what the keyboard shortcut is to activate the feature. You can assign one that you like.

2. In Warp Terminal, ask any question of the AI and when it generates an answer, select the text and hit the shortcut to have it read. Instead of reading the text selected, it will read some other text that's visible on the screen, starting from the top left hand corner. 

3. You can confirm that the feature works in other applications by selecting some text and have it be read. 

### Expected behavior

Ideally you should read the text that's selected. 

### Operating system (OS)

macOS

### Operating system and version

15.5.0

### Current Warp version

v0.2025.06.20.22.47.stable_05

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Does this block you from using Warp daily?

Yes, this issue prevents me from using Warp daily.

### Is this an issue only in Warp?

Yes, I confirmed that this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None
